### PR TITLE
[DOC] Don't suppress autolinks

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -8158,7 +8158,7 @@ rb_ary_deconstruct(VALUE ary)
  *      %i[foo bar baz] # => [:foo, :bar, :baz]
  *      %i[1 % *]       # => [:"1", :%, :*]
  *
- *  - \Method Kernel#Array:
```
Comment (peterzhu2118) [inline array.c:8161]: I think `Method` should be an exception. We don't usually deal with `Method` objects so we shouldn't link to it.

File: array.c [hunk 2]
```diff
@@ -8167,7 +8167,7 @@ rb_ary_deconstruct(VALUE ary)
  *      Array(1)                      # => [1]
  *      Array({:a => "a", :b => "b"}) # => [[:a, "a"], [:b, "b"]]
  *
- *  - \Method Array.new:
```
Comment (peterzhu2118) [inline array.c:8170]: Ditto.

Comment (BurdetteLamar) [reply array.c]: Fixed.

File: array.c (hunk 1 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.  Also, noted in doc guide.

File: doc/contributing/documentation_guide.md [hunk 3]
```diff
@@ -164,6 +164,10 @@ We might consider whether to suppress when:
   (e.g., _Array_ in the documentation for class `Array`).
 - The same reference is repeated many times
   (e.g., _RDoc_ on this page).
+- The reference is to a class or module that users
+  usually don't deal with, including these:
+
+  - \Method.
```
Comment (peterzhu2118) [inline doc/contributing/documentation_guide.md:170]: I think we can also include `Class` and `Module` in this list.

Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md]: Added.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**